internal/httputil/httpapi.go

```diff
@@ -60,6 +61,20 @@ func MakeAuthAPI(
 		logger = logger.WithField("user_id", device.UserID)
 		req = req.WithContext(util.ContextWithLogger(req.Context(), logger))
 
+		// check if the forwarding proxy, if there is one, has provided a real address.
```

Comment:
This _has_ to be gated behind a config option else malicious clients can spoof their IP by literally sending `X-Real-IP: 1.2.3.4` - something like:
```
clientapi.ip_header: X-Forwarded-For
```
might do the trick.